// @ruingl/gptx/src/Types.ts
/*!
 * gptx
 * Copyright (c) 2024 Rui Reogo
 * MIT Licensed
 */

export type ProviderModels = {
    Nextway:
        | 'gpt-4'
        | 'gpt-4o-free'
        | 'gemini-pro'
        | 'gpt-3.5-turbo'
        | 'llama-3.1-405b-instruct-free';

    Aryahcr:
        | 'gpt-4'
        | 'gpt-4-0613'
        | 'gpt-4-32k'
        | 'gpt-4-32k-0314'
        | 'gpt-4-0314'
        | 'gpt-3.5-turbo'
        | 'gpt-3.5-turbo-16k'
        | 'gpt-3.5-turbo-0613'
        | 'gpt-3.5-turbo-16k-0613'
        | 'gpt-3.5-turbo-0301'
        | 'text-davinci-003'
        | 'text-davinci-002'
        | 'code-davinci-002'
        | 'gpt-3'
        | 'text-curie-001'
        | 'text-babbage-001'
        | 'text-ada-001'
        | 'davinci'
        | 'curie'
        | 'babbage'
        | 'ada'
        | 'babbage-002'
        | 'davinci-002';

    Voids:
        | 'gpt-4o-mini-free'
        | 'gpt-4o-mini'
        | 'gpt-4o-free'
        | 'gpt-4-turbo-2024-04-09'
        | 'gpt-4o-2024-08-06'
        | 'grok-2'
        | 'grok-2-mini'
        | 'claude-3-opus-20240229'
        | 'claude-3-opus-20240229-gcp'
        | 'claude-3-sonnet-20240229'
        | 'claude-3-5-sonnet-20240620'
        | 'claude-3-haiku-20240307'
        | 'claude-2.1'
        | 'gemini-1.5-flash-exp-0827'
        | 'gemini-1.5-pro-exp-0827';
};

export interface Options {
    provider: keyof ProviderModels;
    model: ProviderModels[Options['provider']];
}

export interface Message {
    role: 'user' | 'assistant' | 'system';
    content: string;
}
